ksqldb-rest-app/src/test/java/io/confluent/ksql/api/integration/ApiIntegrationTest.java

```diff
@@ -659,6 +678,233 @@ private void shouldFailToExecuteQuery(final String sql, final String message) {
         startsWith(message));
   }
 
+  private void shouldFailToExecutePrint(final String sql, final String message) {
+    // When:
+    PrintResponse printResponse = executePrintTopic(sql);
+    
+    // Then:
+    assertThat(printResponse.rows, hasSize(0));
+    assertThat(printResponse.error.getInteger("error_code"), is(ERROR_CODE_BAD_STATEMENT));
+    assertThat(printResponse.error.getString("message"), startsWith(message));
+  }
+
+  @Test
+  public void shouldExecutePrint() {
+    // Given:
+    String sql = "PRINT " + TEST_TOPIC + " FROM BEGINNING LIMIT 1;";
+
+    // Create a write stream to capture the incomplete response
+    ReceiveStream writeStream = new ReceiveStream(vertx);
+
+    // Make the request to stream a print
+    JsonObject printRequestBody = new JsonObject().put("sql", sql);
+    VertxCompletableFuture<HttpResponse<Void>> responseFuture = new VertxCompletableFuture<>();
+
+    // When:
+    client.post("/query-stream").as(BodyCodec.pipe(writeStream))
+        .sendJsonObject(printRequestBody, responseFuture);
+
+    // Then:
+    AtomicReference<PrintResponse> atomicReference = new AtomicReference<>();
+    assertThatEventually(() -> {
+      try {
+        Buffer buff = writeStream.getBody();
+        PrintResponse printResponse = new PrintResponse(buff.toString());
+        atomicReference.set(printResponse);
+        return printResponse.rows.size();
+      } catch (Throwable t) {
+        return -1;
+      }
+    }, is(1));
+
+    PrintResponse printResponse = atomicReference.get();
+    assertThat(writeStream.isEnded(), is(true));
+    assertThat(printResponse.rows.get(0), containsString(
+        "key: {\"F1\":[\"a\"]}, value: {\"STR\":\"FOO\",\"LONG\":1,\"DEC\":1.11,\"BYTES_\":\"AQ==\",\"ARRAY\":[\"a\"],\"MAP\":{\"k1\":\"v1\"},\"STRUCT\":{\"F1\":2},\"COMPLEX\":{\"DECIMAL\":0.0,\"STRUCT\":{\"F1\":\"v0\",\"F2\":0},\"ARRAY_ARRAY\":[[\"foo\"]],\"ARRAY_STRUCT\":[{\"F1\":\"v0\"}],\"ARRAY_MAP\":[{\"k1\":0}],\"MAP_ARRAY\":{\"k\":[\"v0\"]},\"MAP_MAP\":{\"k\":{\"k\":0}},\"MAP_STRUCT\":{\"k\":{\"F1\":\"v0\"}}},\"TIMESTAMP\":1,\"DATE\":1,\"TIME\":0}, partition: 0"));
+  }
+
+  @Test
+  public void shouldTryToPrintAndSendDoneResponseIfConnectionClosed() {
+    // Given:
+    KsqlEngine engine = (KsqlEngine) REST_APP.getEngine();
+    String sql = "PRINT " + TEST_TOPIC + " FROM BEGINNING LIMIT 1;";
```

Comment:
The fact that this query is FROM BEGINNING with LIMIT means that the connection will be closed even if the client doesn't close it. Can we have this test without the LIMIT clause so that we know for sure that it's testing the client closing the connection, rather than the server?

Reply:
Yes, sure! That's a good point. I changed the sql query to `String sql = "PRINT " + TEST_TOPIC + ";";`. 

Since the connection is closed by the client, we can have any query here, and the test will work. But good point `FROM BEGINNING LIMIT 1` is a bad example to test this, since it will be eventually closed by the server anyway.